### DISCUSSION TITLE
How to disable all tailwind default neutral color shades?

### DISCUSSION THREAD

**Discussion by taylor-xlmedia** [saadeghi/daisyui#3012] [Q&A] [answered]
Forgive me if I missed where to correctly configure something but it seems I'm not able.

My custom theme consists of all the colors defined explicitly:

```
daisyui: {
    themes: [
      {
        myTheme: {
          primary: '#72ff38',
          'primary-focus': '#39c500',
          'primary-content': '#282828',
          secondary: '#ff4924',
          'secondary-focus': '#b11e00',
          'secondary-content': '#fbfbfb',
          accent: '#00f',
          'accent-focus': '#0000b4',
          'accent-content': '#fbfbfb',
          neutral: '#090909',
          'neutral-focus': '#363636',
          'neutral-content': '#fafafa',
          'base-100': '#fff',
          'base-200': '#d9d9d9',
          'base-300': '#4d4d4d',
          'base-content': '#000',
          info: '#7676ff',
          'info-content': '#2b2b73',
          success: '#8dff5f',
          'success-content': '#263a1e',
          warning: '#ff6445',
          'warning-content': '#461e16',
          error: '#ce3515',
          'error-content': '#2b1511',
          '--rounded-box': '0.25rem', // border radius rounded-box utility class, used in card and other large boxes
          '--rounded-btn': '0.25rem', // border radius rounded-btn utility class, used in buttons and similar element
          '--rounded-badge': '1.9rem', // border radius rounded-badge utility class, used in badges and similar
          '--animation-btn': '0.125s', // duration of animation when you click on button
          '--animation-input': '0.2s', // duration of animation for inputs like checkbox, toggle, radio, etc
          '--btn-focus-scale': '0.95', // scale transform of button when you focus on it
          '--border-btn': '1px', // border width of buttons
          '--tab-border': '1px', // border width of tabs
          '--tab-radius': '0.5rem', // border radius of tabs
        },
      },
    ],
  },
  ```

I see all the `neutral` tailwind color shades defined in the source code here:
https://github.com/saadeghi/daisyui/blob/4b7cdb864497122f159ea05fe8d3c817effbe101/src/index.js#L133

I do not want to have these shades be available in my project, but I see no way to remove them.

Love the project - keep up the great work!

**Comment by saadeghi** [comment]
The colors will be in your CSS only if you use their associated utility class names (`bg-neutral-500`, etc)  
Otherwise they won't be in your production CSS file.

**Reply by taylor-xlmedia** [reply]
Thanks for the quick reply.

Yep which is great, but I guess I wanted to disable the neutral shades entirely so my editor does not recommend them for use in the auto complete...

Wondering the thoughts behind the decision to manually reinject those?

**Comment by saadeghi** [comment]
> Wondering the thoughts behind the decision to manually reinject those?

daisyUI had a `neutral` color name since version 1.
Until Tailwind CSS v3, Tailwind [didn't have](https://v2.tailwindcss.com/docs/customizing-colors) a color group called `neutral-*`. It was added in v3 or Tailwind, as a specific shade of gray.  

Because of this name conflict, when daisyUI was adding its colors, it used to simply replace Tailwind's `neutral-*` colors so users couldn't use Tailwind's `neutral-*` colors anymore.  
We couldn't rename the color either because it would break existing websites and there's no better word for it since it's a "neutral" color.

So we had to re-add the Tailwind `neutral-*` colors again so people could use them.

I'm not sure if currently there's a way to remove them again. Let me do some tests…

**Comment by saadeghi** [comment, marked as answer]
> I'm not sure if currently there's a way to remove them again. Let me do some tests…

Here's a way to disable `neutral-*` colors:  
https://play.tailwindcss.com/yjAoOTMHZL?file=config


```js
  theme: {
    extend: {
      colors: {
        'neutral-50': {},
        'neutral-100': {},
        'neutral-200': {},
        'neutral-300': {},
        'neutral-400': {},
        'neutral-500': {},
        'neutral-600': {},
        'neutral-700': {},
        'neutral-800': {},
        'neutral-900': {},
        'neutral-950': {},
      },
    },
  },
```

**Reply by taylor-xlmedia** [reply]
Thanks for digging into this for me. Works great.

Cheers
